aggregator/auth.go

```diff
@@ -185,3 +187,39 @@ func (r *RpcServer) verifyOperator(ctx context.Context, operatorAddr string) (bo
 
 	return auth.VerifyOperator(authRawHeaders[0], operatorAddr)
 }
+
+func (r *RpcServer) GetSignatureFormat(ctx context.Context, req interface{}) (interface{}, error) {
+	walletAddress := ""
+	if reqMap, ok := req.(map[string]interface{}); ok && reqMap["wallet"] != nil {
+		walletAddress = reqMap["wallet"].(string)
+	}
+
+	var chainId *big.Int
+	var err error
+
+	type chainIDGetter interface {
+		ChainID(ctx context.Context) (*big.Int, error)
+	}
+
+	if client, ok := r.ethrpc.(chainIDGetter); ok {
```

Comment:
We definitely do not want to call ethrpc for every GetSignatureFormat(). This value will never change once the aggregator program starts, so during the program initialization phase, this chainId value should already be set. See if there’s a good place to set this global constants during aggregator intialization.